### ISSUE TITLE
sql: Tarantool process hangs after the certain sequencing.

### ISSUE THREAD

**Issue by Hollow111** [tarantool/tarantool#3251]
During the test from the [Tarantool Sql tutorial](https://tarantool.io/en/doc/2.0/tutorials/sql_tutorial.html) the process hanging occurs, thus process can't even be stopped with ctrl+C.

Exact sequencing:
```
box.cfg{}
\set language sql
\set delimiter ;
CREATE TABLE table2 (column1 INTEGER, 
                     column2 VARCHAR(100), 
                     column3 BLOB,  
                     column4 FLOAT, 
                     PRIMARY KEY (column1, column2));
INSERT INTO table2 VALUES (1, 'AB', X'4142', 5.5);
INSERT INTO table2 VALUES (1, 'CD', X'2020', 1E4);
INSERT INTO table2 VALUES (1, 'AB', X'A5', -5.5);
INSERT INTO table2 VALUES (2, 'AB', X'2020', 12.34567);
INSERT INTO table2 VALUES (-1000, '', X'', 0.0);
SELECT column1, column2, column1 * column4 FROM table2 WHERE column2 LIKE '_B';
```

Tarantool, Branch 2.0.

**Comment by pgulutzan** [comment]
I changed the example in the manual so that people won't try these statements while the bug exists.

**Comment by Hollow111** [comment]
**Comparing invalid characters principles in other DMBSs:**

DB2:
- Allows keeping invalid utf-8 characters inside of varchar, but '%'
and '_' in pattern skip any symbols in string regardless of validity.
("SELECT * FROM table1 WHERE smth LIKE '%'" returns all the rows from the table)

reproduce:
```
[[
    CREATE TABLE t(smth VARCHAR(10) PRIMARY KEY) CCSID Unicode
    INSERT INTO t VALUES ('ab' || X'D0')
    SELECT * FROM t WHERE smth
    SELECT * FROM t WHERE smth LIKE ('a%')
    SELECT * FROM t WHERE smth LIKE ('ab%')
]]
```
MySQL:
-  Doesn't let inserting invalid utf-8 characters.
-  But it still allows comparing invalid strings (it doesn't match them though whilst '%' still skips invalid characters).

reproduce:
1
```
[[
    create table t(a char(100));
    insert into t values (convert(0xD0, char(100)));
]]
```
2
```
[[
    select convert(0xD0, char(100)) like convert(0xD0, char(100));
    select convert(0xD0, char(100)) like '%';
]]
```

PostreSQL:
- PostgreSQL doesn't allow using non-utf symbols in string.

reproduce:
```
[[
    SELECT E'\xd0'
]]
```

**Comment by pgulutzan** [comment]
The invalid X'A5' isn't causing the hang, the valid X'' is.

**Comment by Korablev77** [comment]
Closed via 19b64e013e8e7751e04f1916c80a89980ba361cb
